### ISSUE TITLE
ClickHouse: Cannot parse SETTINGS in UNION subqueries

### ISSUE THREAD

**Issue by yuukikuno** [sqlfluff/sqlfluff#7120]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting fails on sections like

```sql
SELECT * FROM test1
UNION ALL
SELECT * FROM test2
SETTINGS allow_experimental_window_functions = 1;
```

### Expected Behaviour

This should parse properly as this is valid.

### Observed Behaviour

```
L:   4 | P:   1 |  PRS | Line 4, Position 1: Found unparsable section: 'SETTINGS
| allow_experimental_window_funct...'
```

### How to reproduce

Run sqlfluff lint --dialect clickhouse filename.sql
filename.sql:

```sql
SELECT * FROM test1
UNION ALL
SELECT * FROM test2
SETTINGS allow_experimental_window_functions = 1;
```

### Dialect

ClickHouse

### Version

3.4.2

### Configuration

[sqlfluff]
templater = placeholder
# Lint regardless of file size
large_file_skip_byte_limit = 0

rules = layout, capitalisation

exclude_rules = capitalisation.functions, capitalisation.identifiers, capitalisation.types

# Supported dialects https://docs.sqlfluff.com/en/stable/perma/dialects.html
# Or run 'sqlfluff dialects'
dialect = clickhouse

# The standard max_line_length is 80 in line with the convention of
# other tools and several style guides. Many projects however prefer
# something a little longer.
# Set to zero or negative to disable checks.
max_line_length = 140

# CPU processes to use while linting.
# The default is "single threaded" to allow easy debugging, but this
# is often undesirable at scale.
# If positive, just implies number of processes.
# If negative or zero, implies number_of_cpus - specified_number.
# e.g. -1 means use all processors but one. 0 means all cpus.
processes = -1

[sqlfluff:indentation]
# While implicit indents are not enabled by default. Many of the
# SQLFluff maintainers do use them in their projects.
allow_implicit_indents = True

[sqlfluff:rules:aliasing.length]
min_alias_length = 3

[sqlfluff:templater:placeholder]
param_style = colon

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
